_posts/2024-08-07-accessing-kafka-with-gateway-api.md

```diff
@@ -0,0 +1,675 @@
+cert-manager-84489bc478-7cxds                       1/1     Running   0          7m22s
+cert-manager-cainjector-7477d56b47-ct8dv            1/1     Running   0          7m22s
+cert-manager-webhook-6d5cb854fc-2rx9p               1/1     Running   0          7m22s
+gateway-api-test-broker-10                          1/1     Running   0          82s
+gateway-api-test-broker-11                          1/1     Running   0          82s
+gateway-api-test-broker-12                          1/1     Running   0          82s
+gateway-api-test-controller-0                       1/1     Running   0          82s
+gateway-api-test-entity-operator-6657fbc775-w4b65   2/2     Running   0          44s
+strimzi-cluster-operator-6948497896-s7q46           1/1     Running   0          2m23s
+```
+
+#### Creating `TLSRoute`s
+
+Next, we will need to create four `TLSRoute`'s:
+
+* A `bootstrap` one that points to the bootstrap service.
+* A `TLSRoute` for each broker that is deployed.
+
+It will be as follows:
+
+```
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: TLSRoute
+metadata:
+  name: gateway-api-test-broker-10
+  namespace: strimzi
+spec:
+  hostnames:
+  - broker-10.strimzi.gateway.api.test
+  parentRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: my-gateway
+    namespace: strimzi
+    sectionName: kafka-listener
+  rules:
+  - backendRefs:
+    - group: ""
+      kind: Service
+      name: gateway-api-test-broker-obiwan-10
+      port: 9092
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: TLSRoute
+metadata:
+  name: gateway-api-test-broker-11
+  namespace: strimzi
+spec:
+  hostnames:
+  - broker-11.strimzi.gateway.api.test
+  parentRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: my-gateway
+    namespace: strimzi
+    sectionName: kafka-listener
+  rules:
+  - backendRefs:
+    - group: ""
+      kind: Service
+      name: gateway-api-test-broker-obiwan-11
+      port: 9092
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: TLSRoute
+metadata:
+  name: gateway-api-test-broker-12
+  namespace: strimzi
+spec:
+  hostnames:
+  - broker-12.strimzi.gateway.api.test
+  parentRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: my-gateway
+    namespace: strimzi
+    sectionName: kafka-listener
+  rules:
+  - backendRefs:
+    - group: ""
+      kind: Service
+      name: gateway-api-test-broker-obiwan-12
+      port: 9092
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: TLSRoute
+metadata:
+  name: gateway-api-test-bootstrap
+  namespace: strimzi
+spec:
+  hostnames:
+  - bootstrap.strimzi.gateway.api.test
+  parentRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: my-gateway
+    namespace: strimzi
+    sectionName: kafka-listener
+  rules:
+  - backendRefs:
+    - group: ""
+      kind: Service
+      name: gateway-api-test-kafka-obiwan-bootstrap
+      port: 9092
+```
+
+```
+kubectl apply -f tls-routes.yaml
```

Comment:
are we following along on github again here? i wonder if we just need to mention that in the first instance.  "If you are following along in GitHub, you just need to apply the YAML configuration files."